src/main/resources/db/migration/R__create_online_merchant_materialized_view.sql

```diff
@@ -0,0 +1,81 @@
+DROP MATERIALIZED VIEW IF EXISTS online_merchant;
+
+CREATE MATERIALIZED VIEW online_merchant AS
+WITH merchant AS (
+    SELECT a.agreement_k,
+           p.name,
+           p.website_url
+    FROM agreement a
+             JOIN profile p ON (p.agreement_fk = a.agreement_k)
+    WHERE a.state = 'APPROVED'
+      AND a.start_date <= CURRENT_TIMESTAMP
+      AND CURRENT_TIMESTAMP <= a.end_date
+      AND p.sales_channel IN ('ONLINE', 'BOTH')
+),
+     product_categories AS (
+         SELECT DISTINCT d.agreement_fk,
+                         pc.product_category
+         FROM discount d
+                  JOIN discount_product_category pc ON (d.discount_k = pc.discount_fk)
+         WHERE d.state = 'PUBLISHED'
+           AND d.start_date <= CURRENT_TIMESTAMP
+           AND CURRENT_TIMESTAMP <= d.end_date
+           AND d.agreement_fk IN (SELECT agreement_k FROM merchant)
```

Comment:
Maybe we can use `EXISTS` clause in this case ?

Reply:
I've switched to EXISTS.
In any case, I've doublechecked the query plans and they are the same 🙂

Reply:
Thanks :)